backend/pkg/httpserver/create_saved_search.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpserver
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+
+	"github.com/GoogleChrome/webstatus.dev/lib/gcpspanner/searchtypes"
+	"github.com/GoogleChrome/webstatus.dev/lib/gen/openapi/backend"
+	"github.com/GoogleChrome/webstatus.dev/lib/httpmiddlewares"
+)
+
+const (
+	savedSearchNameMaxLength            = 32
```

Comment:
Any restriction on the characters in the name?  E.g., is a space allowed? What about punctuation?  Are the case sensitive?  What about uniqueness among a given user's saved searches?

Reply:
> Any restriction on the characters in the name 
No restrictions

> What about uniqueness among a given user's saved searches
Uniqueness of name is not enforced. We don't use the name as an ID. So it would be up to the user if they create multiple searches with the same name.